Assets/LeapMotionModules/UI/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,293 @@
+﻿using Leap.Unity.Attributes;
+using Leap.Unity.RuntimeGizmos;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  public partial class InteractionManager : MonoBehaviour, IRuntimeGizmoComponent {
+
+    [Header("Interactions")]
+    public bool enableHovering = true;
+    public bool enableContact  = true;
+    public bool enableGrasping = true;
+
+    // TODO: Hide these settings behind a dropdown, they probably don't really ever need to be changed
+    [Header("Interaction Settings")]
+    [DisableIf("enableHovering", isEqualTo: false)]
+    public float hoverActivationRadius = 0.5F;
+    [DisableIf("contactOrGraspingEnabled", isEqualTo: false)]
+    public float touchActivationRadius = 0.15F;
+
+    [Header("Layer Settings")]
+    [Tooltip("Whether or not to create the layers used for interaction when the scene runs.")]
+    [SerializeField]
+    protected bool _autoGenerateLayers = true;
+
+    [Tooltip("Layer to use for auto-generation.  The generated interaction layers will have the same collision settings as this layer.")]
+    [SerializeField]
+    protected SingleLayer _templateLayer = 0;
+
+    [Tooltip("The layer containing interaction objects.")]
+    [SerializeField]
+    protected SingleLayer _interactionLayer = 0;
```

Comment:
These layers should be hidden if autoGenerateLaters is enabled, since their serialized value doesn't matter and will be overwritten.  Similarly, the template layer should be hidden if _autoGenerateLayers is false, since it isn't used and isn't important.